packages/web-sdk/src/transports/otlp/transport.ts

```diff
@@ -0,0 +1,174 @@
+import {
+  APIEvent,
+  BaseTransport,
+  createPromiseBuffer,
+  EventEvent,
+  ExceptionEvent,
+  LogEvent,
+  MeasurementEvent,
+  PromiseBuffer,
+  TransportItem,
+  TransportItemType,
+  VERSION,
+} from '@grafana/faro-core';
+import type { TraceEvent } from 'packages/core/src/api';
+import type { OtlpTransportOptions } from './types';
+
+const DEFAULT_BUFFER_SIZE = 30;
+const DEFAULT_CONCURRENCY = 5; // chrome supports 10 total, firefox 17
+
+const DEFAULT_SEND_BATCH_SIZE = 30;
+const DEFAULT_TIMEOUT_MS = 200; // same as default in Otel processor
+const DEFAULT_RATE_LIMIT_BACKOFF_MS = 5000;
+
+export class OtlpTransport extends BaseTransport {
+  readonly name = '@grafana/faro-web-sdk:transport-fetch';
+  readonly version = VERSION;
+
+  private readonly rateLimitBackoffMs: number;
+  private disabledUntil: Date = new Date();
+
+  private readonly getNow: () => number;
+
+  private readonly sendBatchSize: number;
+  private readonly sendTimeoutMs: number;
+
+  private readonly signalsBuffer: TransportItem<APIEvent>[] = [];
+
+  promiseBuffer: PromiseBuffer<Response | void>;
+
+  constructor(private options: OtlpTransportOptions) {
```

Comment:
otel js implementation for traces: https://github.com/open-telemetry/opentelemetry-js/tree/main/experimental/packages/exporter-trace-otlp-http
otel js implementation for metrics: https://github.com/open-telemetry/opentelemetry-js/blob/main/experimental/packages/opentelemetry-exporter-metrics-otlp-http/src/platform/browser/OTLPMetricExporter.ts

Probably hard to incorporate these, but if insipiration is needed :)